### PR TITLE
feat(frontend): show create table

### PR DESCRIPTION
**Pull request by BugenZhao**
I hereby agree to the terms of the [Singularity Data, Inc. Contributor License Agreement](https://gist.github.com/skyzh/0663682a70b0edde7ae991492f2314cb#file-s9y_cla).

## What's changed and what's your intention?

- Unify the normalization for `CREATE` statements, used for `SHOW CREATE` later.
- Store full normalized `CREATE` definition for materialized views and tables.
- Support `SHOW CREATE TABLE`.

We'll use this SQL to retrieve the original definition of the table, for the first version of `ALTER TABLE ADD COLUMN`. 🤣

## Checklist

- [x] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Documentation

If your pull request contains user-facing changes, please specify the types of the changes, and create a release note. Otherwise, please feel free to remove this section.

### Types of user-facing changes

* SQL commands, functions, and operators

### Release note

Support `SHOW CREATE TABLE`.

## Refer to a related PR or issue link (optional)
- #6921 
- #6903 
- #7101

### CODE REVIEW

File: src/frontend/src/handler/create_mv.rs [hunk 1]
```diff
@@ -76,7 +76,7 @@ pub fn gen_create_mv_plan(
 
     let (database_id, schema_id) = session.get_database_and_schema_id_for_create(schema_name)?;
 
-    let definition = query.to_string();
```
Comment (BugenZhao) [inline src/frontend/src/handler/create_mv.rs:79]: With this previous method, `WITH` options cannot be preserved. That'll be problematic for sinks.

File: src/frontend/src/handler/show.rs [hunk 2]
```diff
@@ -176,7 +175,7 @@ pub fn handle_show_create_object(
 
     Ok(PgResponse::new_for_stream(
         StatementType::SHOW_COMMAND,
-        Some(1),
```
Comment (yezizp2012) [inline src/frontend/src/handler/show.rs:179]: The row count should be 1? 🤔

File: src/frontend/src/catalog/table_catalog.rs [hunk 3]
```diff
@@ -107,7 +107,7 @@ pub struct TableCatalog {
     /// is not supported yet and expected to be `[0..columns.len()]`
     pub value_indices: Vec<usize>,
 
-    /// Definition of the materialized view.
+    /// The full `CREATE TABLE` or `CREATE MATERIALIZED VIEW` definition of the table.
```
Comment (yezizp2012) [inline src/frontend/src/catalog/table_catalog.rs:110]: The `definition` is also used in system table `pg_matviews`, which was already used by the cloud team. I'm not sure how they are using it. We'd better check with them first.

**Review by yezizp2012 — COMMENTED** [review]
Good job, rest LGTM.

File: src/frontend/src/catalog/table_catalog.rs (hunk 3 above)
Comment (yezizp2012) [reply src/frontend/src/catalog/table_catalog.rs:110]: https://github.com/risingwavelabs/risingwave-cloud/blob/2c28ddead1cab0a298c314d9ab8a0119be520e81/mgmt/internal/model/mat_view.go#L238

Comment (BugenZhao) [reply src/frontend/src/catalog/table_catalog.rs:110]: https://github.com/risingwavelabs/risingwave-cloud/pull/844

Looks like the definition is only used for representing to users in a highlight style. cc @Ad-Bean

File: src/frontend/src/handler/show.rs (hunk 2 above)
Comment (BugenZhao) [reply src/frontend/src/handler/show.rs:179]: This field seems only used when `StatementType` is not a query. 🤔

https://github.com/risingwavelabs/risingwave/blob/8d082ecfc9d1c93bc7c6ef87dfeca60f381250d5/src/utils/pgwire/src/pg_response.rs#L131-L139

Comment (yezizp2012) [reply src/frontend/src/handler/show.rs:179]: Make sense. Please help to fix it in other handing parts of  non-queries.

**Review by chenzl25 — APPROVED** [review]
LGTM. A quick question, if we add a column to a table, how to modify the definition of the table? `original definition` + `add column statement` or regenerate a new definition.

**Review by ZENOTME — APPROVED** [review]

**Review by yezizp2012 — APPROVED** [review]

**Comment by BugenZhao** [comment]
> regenerate a new definition.

I guess to show this one, according to the behavior of MySQL. 🤔 It always frankly shows how to create an empty table with the same schema in one-liner. BTW, for `create table as`, we should output `create table (v int, v2 int)` as well. 🥵

**Pull request merged**